splitio/sync/split.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, split_api, split_storage):
                                 _ON_DEMAND_FETCH_BACKOFF_BASE,
                                 _ON_DEMAND_FETCH_BACKOFF_MAX_WAIT)
 
-    def _fetch_until(self, fetch_options, till=None):
+    def _fetch_until(self, fetch_options, till=None, segment_sync=None):
```

Comment:
segment sync should not be optional since it's used without checks.
still, i'd suggest not passing around the segment sync so deep and just returning a list of all the segments referenced in the splits. later on, when the segments are indeed being fetched, filter out the ones that are already present

Reply:
The logic used to be in sync/synchronizer.py, Matti asked to move it to sync/split.py, that is why segment_sync object is needed. I will put a check before using it to avoid errors.